packages/components/src/test/properties/NavigationPropertyTargetSelector.test.tsx

```diff
@@ -168,16 +160,153 @@ describe("NavigationPropertyTargetSelector", () => {
+
+    const initialProps: NavigationPropertyTargetSelectorProps = {
+      imodel: testImodel,
+      getNavigationPropertyInfo: async () => testNavigationPropertyInfo,
+      propertyRecord,
+    };
+    const { container, getByDisplayValue, getByRole, queryByText } = render(<NavigationPropertyTargetSelector {...initialProps} />);
+
+    const dropdownButton = await waitFor(() => {
+      const element = container.querySelector<HTMLDivElement>(".iui-end-icon");
+      expect(element).to.not.be.null;
+      return element;
+    });
+
+    // when input value is empty
+    fireEvent.click(dropdownButton!);
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.not.be.null);
+    fireEvent.click(dropdownButton!);
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.be.null);
+    expect((getByRole("textbox") as HTMLInputElement).value).to.be.eq("");
+
+    // when input value is not empty
+    fireEvent.click(dropdownButton!);
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.not.be.null);
+    const menuItem = queryByText(contentItem.label.displayValue);
+    fireEvent.click(menuItem!);
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.be.null);
+    expect(getByDisplayValue(contentItem.label.displayValue)).to.not.be.null;
+  });
+
+  it("correctly handles keyDown events", async () => {
+    const user = userEvent.setup();
+    sinon.stub(Presentation.presentation, "getContent").resolves(new Content(createTestContentDescriptor({ fields: [], categories: [] }), [contentItem]));
+    const propertyDescription = createNavigationPropertyDescription();
+    const value = {
+      valueFormat: PropertyValueFormat.Primitive,
+      value: { id: "1", className: "TestSchema:TargetClass" },
+      displayValue: "Target Class Instance",
+    };
+    const propertyRecord = new PropertyRecord(value as PropertyValue, propertyDescription);
+
+    const initialProps: NavigationPropertyTargetSelectorProps = {
+      imodel: testImodel,
+      getNavigationPropertyInfo: async () => testNavigationPropertyInfo,
+      propertyRecord,
+    };
+    const { getByDisplayValue, getByRole, queryByText } = render(<NavigationPropertyTargetSelector {...initialProps} />);
+
+    const inputContainer = await waitFor(() => getByRole("textbox"));
+
+    fireEvent.click(inputContainer);
+    await user.keyboard("{End}");
+    await user.type(inputContainer, "E", { skipClick: true });
+
+    await user.keyboard("{Home}");
+    await user.type(inputContainer, "H ", { skipClick: true });
+
+    await waitFor(() => expect(getByDisplayValue("H Target Class InstanceE")).to.not.be.null);
+
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.not.be.null);
+
+    fireEvent.keyDown(inputContainer, { key: "Tab" });
+    await waitFor(() => expect(queryByText(contentItem.label.displayValue)).to.be.null);
```

Comment:
I think it would be worth adding comment to explain what you are checking. e.g. 
```
// check if menu was closed after `tab` key was pressed.
```

Same in lines below where `E` is typed and checking if menu is opened.

Reply:
added comments